Change the name of `-o` flag to `-overlap` flag

vs74 opened the pull request:
## Checklist

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [x] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've added relevant tests for my contribution
- [x] I've updated the documentation and/or added correct docstrings
- [x] I've assigned a reviewer
- [x] I've consulted [ADS's internal developer documentation](https://github.com/neuropoly/axondeepseg/wiki/Guidelines-for-contribution) to ensure my contribution is in line with any relevant design decisions


<!--- For the title, please observe the following rules:
	- Provide a concise and self-descriptive title
	- Do not include the applicable issue number in the title, do it in the PR body
	- If the PR is not ready for review, convert it to a draft.
-->

## Description
As pointed out by Julien in the issue https://github.com/neuropoly/axondeepseg/issues/460, the `-o` flag for segmenting the images sounds like output, in this PR I've dropped the single dash overlap `-o` flag which is confusing.

## How to test this PR?
1. First, do `pip install -e .` in this branch.
2. Segment an image using changed flag name; `axondeepseg -i <image_name.png> -t <Modality type> --overlap <overlap_value>`

## Changed Documentation of this PR
https://axondeepseg--474.org.readthedocs.build/en/474/documentation.html#syntax

## Linked issues
Fixes #460

mathieuboudreau commented:
I *think* single dash CLI flags are usually meant for single letters. I don't see the point of using `-overlap` if we already also have `--overlap`.

Thoughts from anyone else?

vs74 commented:
In some RTDs, I've seen the practice of single dashed even for words. That's why I added that. If it is true that single dashed are only for letters then perhaps we can remove the `-o` flag ( in CLI's of several other packages, `-o` flag usually means 'output'  so it can create confusion among the users) and keep the double dashed `--overlap` flag of the CLI.

jcohenadad commented:
> I think single dash CLI flags are usually meant for single letters. I don't see the point of using -overlap if we already also have --overlap.

indeed-- single letter after "-" is the POSIX convention.  I was not aware we already had "--overlap", in this case we can simply drop "-o". In general i'm in favour of getting rid of single dash single letter (becomes confusing).

For some context. It all started (in the lab, inspired by FSL) with SCT using single dash with long words, and no double-dash. We are having this discussion regularly in other projects. Recently for shimming toolbox we decided to stick to POSIX (eg double dash for long words), but avoid single dash. Few discussions:
- https://github.com/shimming-toolbox/shimming-toolbox/issues/222
- https://github.com/neuropoly/spinalcordtoolbox/issues/2939
- https://github.com/neuropoly/spinalcordtoolbox/issues/3159

mathieuboudreau commented:
Perfect - thank you for clarifying the new lab convention @jcohenadad !

So @vs74, I guess all that needs to be done in this PR is removing the `-o` flag and then it can be merged, correct?

vs74 commented:
For now, I've dropped the `-o` flag, and this PR is ready for review. I've dropped the -o flag, changed the documentation added a unit test that was missing for the overlap flag. But we should open an issue too just for the case that if in future we wish to switch to single dash or double-dash entirely for CLI flags, we could refer to that issue.

mathieuboudreau reviewed (approved):
Looks great - once the test pass I'll merge!
